### PR TITLE
Start phone api

### PR DESCRIPTION
**Pull request by groovecoder**
How to test:
Go thru the new `docs/end-to-end-local-phone-dev.md` instructions!

- [x] I've added a unit test to test for potential regressions of this bug.
- [x] I've added or updated relevant docs in the docs/ directory.

### CODE REVIEW

File: docs/end-to-end-local-phone-dev.md
Comment (jwhitlock) [inline docs/end-to-end-local-phone-dev.md:83]: Thanks, this got me to find the docs for [ngrok Agent Config](https://ngrok.com/docs/ngrok-agent/config)

File: api/views.py
Comment (jwhitlock) [inline api/views.py:235]: Suggestion: `datetime.now(timezone.utc)` for timezone-aware dates. Seeing warnings in server logs:

```
/Users/john/.pyenv/versions/3.9.11/lib/python3.9/site-packages/django/db/models/fields/__init__.py:1416: RuntimeWarning: DateTimeField RealPhone.verification_sent_date received a naive datetime (2022-06-27 20:29:43.103116) while time zone support is active.
  warnings.warn("DateTimeField %s received a naive datetime (%s)"
```

File: phones/models.py
Comment (jwhitlock) [inline phones/models.py:24]: Suggestion: `datetime.now(timezone.utc)` for timezone-aware dates. Seeing warnings in server logs:

```
/Users/john/.pyenv/versions/3.9.11/lib/python3.9/site-packages/django/db/models/fields/__init__.py:1416: RuntimeWarning: DateTimeField RealPhone.verification_sent_date received a naive datetime (2022-06-27 20:29:43.103116) while time zone support is active.
  warnings.warn("DateTimeField %s received a naive datetime (%s)"
```

File: phones/models.py
Comment (jwhitlock) [inline phones/models.py:23]: Suggest changing the name:

```suggestion
def verification_sent_date_default():
```

This will have to be reflected several other places, like the migration.

File: docs/end-to-end-local-phone-dev.md
Comment (jwhitlock) [inline docs/end-to-end-local-phone-dev.md:116]: I believe additional steps are needed, something like:

1. Go to https://console.twilio.com/us1/develop/phone-numbers/manage/incoming
2. Select phone number to edit
3. Under Configure Tab (default), scroll to "Voice & Fax"
4. In "Configure With", change from "Webhook, TwiML Bin, Function, Studio Flow, Proxy Service" to "TwiML App"
5. Set TWIML APP to the newly-created app
6. Scroll to "Messaging"
7. In "Configure With", change from "Webhook, TwiML Bin, Function, Studio Flow, Proxy Service" to "TwiML App"
8. Set TWIML APP to the newly-created app
9. Select "Save" to apply

Now when you text the app number, it causes a `POST /api/v1/inbound_sms`.

File: docs/end-to-end-local-phone-dev.md
Comment (groovecoder) [reply docs/end-to-end-local-phone-dev.md:61]: The main thing I'm going to take away from this is to just remove all of the old "proxy" phone code completely.

File: docs/end-to-end-local-phone-dev.md
Comment (groovecoder) [reply docs/end-to-end-local-phone-dev.md:71]: Added.

File: docs/end-to-end-local-phone-dev.md
Comment (groovecoder) [reply docs/end-to-end-local-phone-dev.md:116]: I added a whole new section after this for `Set up a local Relay number to receive and forward messages`. It describes how to use the couple of http://127.0.0.1:8000/api/v1/docs/ endpoints to buy a relay number and send a text message thru it and ngrok.

File: docs/end-to-end-local-phone-dev.md
Comment (groovecoder) [reply docs/end-to-end-local-phone-dev.md:62]: I removed the `SERVICE` env var completely.

**Comment by groovecoder** [comment]
@jwhitlock , @say-yawn - I'd like to merge this now. Some of the safeguards to make sure it doesn't load on an environment that's not ready yet. (E.g., stage, prod)

1. I completely removed the previous `/phones/*` urls and views
2. Django doesn’t load the phones app unless `TWILIO_ACCOUNT_SID` and `TWILIO_AUTH_TOKEN` env vars are set
3. The API app doesn’t add any phones viewsets or urls unless the phones app is loaded

I've also added significant tests too!

File: api/views.py
Comment (groovecoder) [inline api/views.py:426]: Note: we may want to throttle this API endpoint too.

File: api/views.py
Comment (jwhitlock) [inline api/views.py:259]: Should we detect if a `RealPhone` has a still-valid verification code and skip creating and sending a new one? I could see this being used to:

1) Send 1,000s of SMS messages to a target phone, to annoy them
2) Generate 10,000 verification codes, to increase the chances of brute forcing

**Review by jwhitlock — CHANGES_REQUESTED** [review]
Thanks @groovecoder! I was able to setup SMS relaying on my personal Twilio account. I think we could merge after a few more rounds, and track follow-on tasks in new issues and PRs.

There's a few have-to-fix items, some (mostly ignorable) suggestions, and some nits. Please re-request a review when you want me to look at updates.

File: api/renderers.py
Comment (jwhitlock) [inline api/renderers.py:1]: If you add `"vobject"` and `"twilio.request_validator"` to the ignored modules list, `mypy` will like your PR.

File: docs/end-to-end-local-phone-dev.md
Comment (groovecoder) [reply docs/end-to-end-local-phone-dev.md:151]: That all sounds like the intended behavior. I checked with Product that we can go with this behavior but we'll be open to changing it later.

File: .coveragerc
Comment (groovecoder) [reply .coveragerc:9]: Yeah, it found a `.py` file in there. Added `env/*`

File: phones/admin.py
Comment (groovecoder) [reply phones/admin.py:3]: Yup. Removed.

File: phones/migrations/0006_realphone.py
Comment (groovecoder) [reply phones/migrations/0006_realphone.py]: I ran it on the migrations but I'm going to hold off `black` formatting everything until I get thru more code changes. I think I'd much rather have a single commit or follow-up PR that is like "black format all the phone code" if that's alright?

File: phones/models.py
Comment (groovecoder) [reply phones/models.py]: Oh dang yeah that'd be a huge index for not much gain. Good catch! Removed.

File: privaterelay/middleware.py
Comment (groovecoder) [reply privaterelay/middleware.py:73]: Yes, just to make it easier for testing different values of it.

File: api/views.py
Comment (groovecoder) [reply api/views.py:386]: Hmm ... I can't think of a way we can know for sure if `+1503` meaning "country code 1 (US) area code 503" and `+503` meaning "country code 503 (El Salvador)". I.e., "+503xxxyyyy" *IS* a valid e.164 number format.

Also, this `_get_number_details` call failure is more generic like "Twilio couldn't return number details" so it might not have anything to do with e.164. So I suggest just keeping the error message as-is.

File: phones/models.py
Comment (groovecoder) [reply phones/models.py:88]: I added another check and a test for this.

File: phones/models.py
Comment (groovecoder) [reply phones/models.py:166]: I removed the TODO and made it TODONE in the code.

File: phones/models.py
Comment (groovecoder) [reply phones/models.py:168]: I removed the TODO and made it TODONE in the code.

**Review by jwhitlock — APPROVED** [review]
Thanks @groovecoder, this looks like a solid first version.

Before merging, please:
* Reformat with `black`
* Write a follow-on issue for relay logo in vCard (or add as TODO for PR #2171)

**Review by jwhitlock — APPROVED** [review]

**Pull request merged**
